TRADEOFFS.md

```diff
@@ -0,0 +1,18 @@
+# Tradeoffs
+
+
+## Potential reasons to use React
+
+- React has a HUGE communitity! If you have a problem or need odds are you'll be able to find someone who shares your needs and may already have a solution!
+- Create React App! Is a great way to start a React project without needing to futz around with tooling.
+
+## Potential reasons to not use React
+
+- Most React apps typically require a build step, this isn't technically required, but you'll find that the vast majority of React apps have one, and all learning material assumes one.
+- It is JavaScript-centric. If you aren't interested in learning JavaScript well, React may be harder to work with than template based libraries.
+- Larger runtime than other options. For projectes sensitive to byte size, such as embedded widgets, React may be too big
+- React's performance is worse than template-based solutions for cases where many things update extremely quickly at the same time, such as a stock trading app.
```

Comment:
is this is always the case? can we elaborate more about this?

Reply:
I think this is because you can compile template directly into DOM calls, so there is no overhead of diffing (of course the performance is always tricky but in general, compiled templates are faster - see marko for example)

Reply:
I see, so in other words, react needs to traverse the component tree and compare the difference every time it renders, and a compiled template can be optimized ahead of time to just modify the DOM directly? For example:

```
<div>
  <div>
    {props.children}
  </div>
</div>
```

if we want to update the props.children,

In React this would be a minimum of 2 createElement calls at runtime and then apply the difference to the DOM (two steps that happen for every render)

In an AoT template, this can be just one step (to put it simply), because the compiler knows that the only thing that needs to be updated is the `props.children`. Is that right?

Reply:
Yes :)

Reply:
This also affects the initial render time, because you first must build the `createElement` tree, then diff it against the DOM (which should be missing everything), then commit it.

This is mitigated by doing server side rendering, of course.